service/history/workflow_task_handler_callbacks_update_test.go

```diff
@@ -0,0 +1,410 @@
+		workflowCache      wcache.Cache
+		mockNamespaceCache *namespace.MockRegistry
+
+		logger log.Logger
+
+		workflowTaskHandlerCallback *workflowTaskHandlerCallbacksImpl
+	}
+)
+
+func TestWorkflowTaskHandlerCallbacksUpdateSuite(t *testing.T) {
+	suite.Run(t, new(WorkflowTaskHandlerCallbacksUpdateSuite))
+}
+
+func (s *WorkflowTaskHandlerCallbacksUpdateSuite) SetupTest() {
+	s.Assertions = require.New(s.T())
+	s.ProtoAssertions = protorequire.New(s.T())
+	s.HistoryRequire = historyrequire.New(s.T())
+	s.UpdateUtils = updateutils.New(s.T())
+
+	s.controller = gomock.NewController(s.T())
+	config := tests.NewDynamicConfig()
+	mockShard := shard.NewTestContext(
+		s.controller,
+		&persistencespb.ShardInfo{
+			ShardId: 1,
+			RangeId: 1,
+		},
+		config,
+	)
+	mockShard.Resource.ShardMgr.EXPECT().AssertShardOwnership(gomock.Any(), gomock.Any()).AnyTimes()
+	mockShard.Resource.ShardMgr.EXPECT().UpdateShard(gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+
+	s.mockNamespaceCache = mockShard.Resource.NamespaceCache
+	s.mockExecutionMgr = mockShard.Resource.ExecutionMgr
+	mockClusterMetadata := mockShard.Resource.ClusterMetadata
+	mockClusterMetadata.EXPECT().GetCurrentClusterName().Return(cluster.TestCurrentClusterName).AnyTimes()
+	mockClusterMetadata.EXPECT().ClusterNameForFailoverVersion(false, common.EmptyVersion).Return(cluster.TestCurrentClusterName).AnyTimes()
+	mockClusterMetadata.EXPECT().ClusterNameForFailoverVersion(true, tests.Version).Return(cluster.TestCurrentClusterName).AnyTimes()
+	mockClusterMetadata.EXPECT().GetAllClusterInfo().Return(nil).AnyTimes()
+
+	s.mockEventsCache = mockShard.MockEventsCache
+	s.mockEventsCache.EXPECT().PutEvent(gomock.Any(), gomock.Any()).AnyTimes()
+	s.logger = mockShard.GetLogger()
+
+	s.workflowCache = wcache.NewHostLevelCache(mockShard.GetConfig(), metrics.NoopMetricsHandler)
+
+	mockTxProcessor := queues.NewMockQueue(s.controller)
+	mockTimerProcessor := queues.NewMockQueue(s.controller)
+	mockVisibilityProcessor := queues.NewMockQueue(s.controller)
+	mockArchivalProcessor := queues.NewMockQueue(s.controller)
+	mockMemoryScheduledQueue := queues.NewMockQueue(s.controller)
+	mockTxProcessor.EXPECT().Category().Return(tasks.CategoryTransfer).AnyTimes()
+	mockTimerProcessor.EXPECT().Category().Return(tasks.CategoryTimer).AnyTimes()
+	mockVisibilityProcessor.EXPECT().Category().Return(tasks.CategoryVisibility).AnyTimes()
+	mockArchivalProcessor.EXPECT().Category().Return(tasks.CategoryArchival).AnyTimes()
+	mockMemoryScheduledQueue.EXPECT().Category().Return(tasks.CategoryMemoryTimer).AnyTimes()
+	mockTxProcessor.EXPECT().NotifyNewTasks(gomock.Any()).AnyTimes()
+	mockTimerProcessor.EXPECT().NotifyNewTasks(gomock.Any()).AnyTimes()
+	mockVisibilityProcessor.EXPECT().NotifyNewTasks(gomock.Any()).AnyTimes()
+	mockArchivalProcessor.EXPECT().NotifyNewTasks(gomock.Any()).AnyTimes()
+	mockMemoryScheduledQueue.EXPECT().NotifyNewTasks(gomock.Any()).AnyTimes()
```

Comment:
Maybe we should reorder lines, so each mock creation is followed by expectations.  Since you are not asserting any specific aruguments nor how many times methods can be called the proper name for the entity is stub. There are number of different test doubles and gomock can be used as many of them.

Reply:
I copied it from other test suite. Consistency is more important than personal preferences.